hazelcast/src/main/java/com/hazelcast/config/CredentialsFactoryConfig.java

```diff
@@ -72,47 +75,51 @@ public CredentialsFactoryConfig setProperties(Properties properties) {
         return this;
     }
 
+    @Override
+    public ICredentialsFactory asCredentialsFactory(ClassLoader cl) {
+        if (implementation != null) {
+            return implementation;
+        } else {
+            try {
+                ICredentialsFactory credentialsFactory = ClassLoaderUtil.newInstance(cl, className);
+                credentialsFactory.init(properties);
```

Comment:
Is it expected that we will create `credentialsFactory` every time? Maybe we should cache it?